### DISCUSSION TITLE
Subscribing To Changes To User Record Help

### DISCUSSION THREAD

**Discussion by labthe3rd** [pocketbase/pocketbase#3262] [Q&A]
I'm attempting to track user changes to update the my nextJS application when a user makes any changes to their account. To do this I created a hook to subscribe to the user's record via the user's ID. However when I attempt to use pb.authstore.clear() I am met with the Unhandled Runtime Error
ClientResponseError 404: Missing or invalid client id.

I am not sure what I am doing wrong. I have found that commenting out the subscribe and unsubscribe lines, the problem goes away but then I am not tracking the user data being changed. Below is my hook that I am using, but if more info is needed I will gladly provide anything to get some assistance. Is there something I am doing wrong or a better way to go about this?

```
/src/hooks/useSubscribeUser.js

import { useEffect } from "react";
import pb from "@/lib/pocketbase";

const useSubscribeUser = (userId, setUser) => {
  useEffect(() => {
    if (!userId) return;

    const handleEvent = (e) => {
      if (e.action === "update") {
        if (e.record) {
          setUser(e.record);
        }
      }
    };

    console.log("Subscribing user with id: ", userId);
    pb.collection("users").subscribe(userId, handleEvent);

    return () => {
      pb.collection("users").unsubscribe(userId);
    };
  }, [userId, setUser]);
};

export default useSubscribeUser;
```

**Comment by ganigeorgiev** [comment]
I'm not sure I follow.

Where the above code is running - client-side or server-side? If the latter note that it is not safe to have a global `pb` instance. I assume that it is client-side due to the `useEffect`, right?

Also note that `pb.authStore.clear()` doesn't perform any requests - it only resets the store's `pb.authStore.token` and `pb.authStore.model` values.

The `404: Missing or invalid client id` could be a result from the unsubscribe call if the client was already previously disconnected/removed.

**Reply by ganigeorgiev** [reply]
You can also try to start the PocketBase executable with `--debug` to check in the console for the realtime connection messages for your client and whether they are executed in the order you expect.

**Comment by labthe3rd** [comment]
My apologies, I should have specified, I am using pocket base in a nextJS/React project. Thank you for the response. I am running the hook on the client side as a useEffect. In my header component I have a logout button that performs the pb.authstore.clear() which is triggering the use effect that attempts to unsubscribe. Everything appears to work properly besides the error popping up, but it seems to be harmless since it is just attempting to run an unsubscribe with pb.authstore being cleared. I’ll try to add some more checks to prevent the hook from attempting to unsubscribe after the logout button is pressed. Sent from my iPhoneOn Sep 5, 2023, at 2:35 PM, Gani Georgiev ***@***.***> wrote:﻿
I'm not sure I follow.
Where the above code is running - client-side or server-side? If the latter note that it is not safe to have a global pb instance.
Also note that pb.authstore.clear() doesn't perform any requests - it only resets the store's pb.authStore.token and pb.authStore.model values.
The 404: Missing or invalid client id could be a result from the unsubscribe call if the subscription was already previously removed.
